pwiz_tools/Skyline/Controls/Graphs/AreaReplicateGraphPane.cs

```diff
@@ -1,1473 +1,1518 @@
-
-                    TextObj textObj = new TextObj(text,
-                                                  i + FirstDataIndex + 1, sumAreas[i],
-                                                  CoordType.AxisXYScale,
-                                                  AlignH.Center,
-                                                  AlignV.Bottom)
-                                          {
-                                              IsClippedToChartRect = true,
-                                              ZOrder = ZOrder.F_BehindGrid,
-                                          };
-
-
-                    textObj.FontSpec.Border.IsVisible = false;
-                    textObj.FontSpec.Size = pointSize.Value;
-                    _labelHeight =(int) textObj.FontSpec.GetHeight(CalcScaleFactor());
-                    GraphObjList.Add(textObj);
-                    _dotpLabels.Add(textObj);
-                }
-            }
-        }
-
-        private int? GetDotProductsPointSize(Graphics g)
-        {
-            for (int pointSize = (int) Settings.Default.AreaFontSize; pointSize > 4; pointSize--)
-            {
-                var fontLabel = new FontSpec {Size = pointSize};
-                var sizeLabel = fontLabel.MeasureString(g, DotpLabelText, CalcScaleFactor());
-
-                float labelWidth = (float) XAxis.Scale.ReverseTransform((XAxis.Scale.Transform(0) + sizeLabel.Width));
-
-                if (labelWidth < 1.2)
-                    return pointSize;          
-            }
-            return null;
-        }
-
-        private string GetDotProductResultsText(int indexResult)
-        {
+                _dotpData = ImmutableList.ValueOf(graphData.DotpData.Select(point => (float)point.Y));
+
+            if (ExpectedVisible != AreaExpectedValue.none &&
+                DotProductDisplayOption.line.IsSet(Settings.Default))
+            {
+                Y2Axis.IsVisible = true;
+                Y2Axis.Scale.Min = 0;
+                Y2Axis.Scale.Max = 1.1;
+                Y2Axis.Title.Text = DotpLabelText;
+                var dotpLine = new LineItem(DotpLabelText, graphData.DotpData, Color.DimGray, SymbolType.Circle )
+                {
+                    IsY2Axis = true, Line = new Line() { Style = DashStyle.Dash, Color = Color.DimGray, Width = 2.0f},
+                    Symbol = new Symbol() { Type = SymbolType.Diamond, Size = 5f, Fill = new Fill(Color.DimGray)}
+                };
+                CurveList.Insert(0, dotpLine);
+                ToolTip.TargetCurves.ClearAndAdd(dotpLine);
+            }
+            else
+            {
+                Y2Axis.IsVisible = false;
+            }
+
+            UpdateAxes(resetAxes, aggregateOp, dataScalingOption, normalizeOption);
+
+            if (Settings.Default.PeakAreaDotpCutoffShow && DotProductDisplayOption.line.IsSet(Settings.Default) && _dotpData != null)
+            {
+                var cutoff = Settings.Default.PeakAreaDotpCutoffValue;

```

Comment:
Realistically, the user may want a different cutoff for each of the 3 different types: dotp, idotp, rdotp. Their reliability of the target values are different in the different cases.

Reply:
Fixed